docs/book/src/developer/providers/v1alpha3-to-v1alpha4.md

```diff
@@ -245,3 +245,10 @@ with `cert-manager.io/v1`
   ```yaml
   serviceAccountName: manager
   ```
+
+## Percentage String or Int API input will fail with a string different from an integer with % appended.
+
+`MachineDeployment.Spec.Strategy.RollingUpdate.MaxSurge`, `MachineDeployment.Spec.Strategy.RollingUpdate.MaxUnavailable` and `MachineHealthCheck.Spec.MaxUnhealthy` would have previously taken a String value with an integer character in it e.g "3" as a valid input and process it as a percentage value.
+Only String values like "3%" or Int values e.g 3 are valid input values now. A string not matching the percentage format will fail, e.g "3".
```

Comment:
That's probably going to convert `3` from v1alpha4 to `3%` all the time there is a conversion, because the conversion code cannot tell if it was a percentage already or not.

Reply:
Wha about 
if isString && !HasSuffix("%"), then add the suffix

Reply:
This would also open the discussion of should "3" convert to "3%" or to 3, so I'd personally be in favour of not adding any magic.

Reply:
`if isString && !HasSuffix("%"), then add the suffix`

As @enxebre mentioned, we wouldn't know if the "3" value is coming from an already-v1alpha4 object and convert that to a percentage on their behalf (which would be a bug)